internal/integrationtest/sketch/sketch_test.go

```diff
@@ -0,0 +1,887 @@
+	defer env.CleanUp()
+
+	_ = cli.CopySketch("sketch_simple")
+	// Creates a folder where to save the zip
+	archivesFolder := cli.WorkingDir().Join("my_archives")
+	require.NoError(t, archivesFolder.Mkdir())
+
+	_, _, err := cli.Run("sketch", "archive", cli.WorkingDir().Join("sketch_simple").String(), "./my_archives/my_custom_sketch.zip")
+	require.NoError(t, err)
+
+	archive, err := zip.OpenReader(cli.WorkingDir().Join("my_archives", "my_custom_sketch.zip").String())
+	require.NoError(t, err)
+	defer require.NoError(t, archive.Close())
+	verifyZipContainsSketchExcludingBuildDir(t, archive.File)
+}
+
+func TestSketchArchiveAbsoluteSketchPathWithAbsoluteZipPathAndNameWithoutExtension(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	_ = cli.CopySketch("sketch_simple")
+	// Creates a folder where to save the zip
+	archivesFolder := cli.WorkingDir().Join("my_archives")
+	require.NoError(t, archivesFolder.Mkdir())
+
+	_, _, err := cli.Run("sketch", "archive", cli.WorkingDir().Join("sketch_simple").String(), archivesFolder.Join("my_custom_sketch").String())
+	require.NoError(t, err)
+
+	archive, err := zip.OpenReader(archivesFolder.Join("my_custom_sketch.zip").String())
+	require.NoError(t, err)
+	defer require.NoError(t, archive.Close())
+	verifyZipContainsSketchExcludingBuildDir(t, archive.File)
+}
+
+func TestSketchArchiveAbsoluteSketchPathWithAbsoluteZipPathAndNameWithExtension(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	_ = cli.CopySketch("sketch_simple")
+	// Creates a folder where to save the zip
+	archivesFolder := cli.WorkingDir().Join("my_archives")
+	require.NoError(t, archivesFolder.Mkdir())
+
+	_, _, err := cli.Run("sketch", "archive", cli.WorkingDir().Join("sketch_simple").String(), archivesFolder.Join("my_custom_sketch.zip").String())
+	require.NoError(t, err)
+
+	archive, err := zip.OpenReader(archivesFolder.Join("my_custom_sketch.zip").String())
+	require.NoError(t, err)
+	defer require.NoError(t, archive.Close())
+	verifyZipContainsSketchExcludingBuildDir(t, archive.File)
+}
+
+func TestSketchArchiveNoArgsWithIncludeBuildDirFlag(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	cli.SetWorkingDir(cli.CopySketch("sketch_simple"))
+	_, _, err := cli.Run("sketch", "archive", "--include-build-dir")
+	require.NoError(t, err)
+	cli.SetWorkingDir(env.RootDir())
+
+	archive, err := zip.OpenReader(cli.WorkingDir().Join("sketch_simple.zip").String())
+	require.NoError(t, err)
+	defer require.NoError(t, archive.Close())
+	verifyZipContainsSketchIncludingBuildDir(t, archive.File)
+}
+
+func TestSketchArchiveDotArgWithIncludeBuildDirFlag(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	cli.SetWorkingDir(cli.CopySketch("sketch_simple"))
+	_, _, err := cli.Run("sketch", "archive", ".", "--include-build-dir")
```

Comment:
From here all the `archive` tests repeat with the `--include-build-dir` flag added, this can be automated.